RIOT-OS/RIOT
# boards: examples: tests: refactor the way boards providing netif is defined

### Contribution description

This PR is RFC because I'm not 100% sure of the approach but the idea behind is to get rid of the `BOARDS_PROVIDES_NETIF` variables in `examples/default` and `tests/netstat_l2` applications by using a specific feature for boards providing an interface (using `FEATURES_REQUIRED += <something>`). The `<something>` should be generic enough to cover 802154, ble, lora, wifi, ethernet, etc and must be provided at board level, similar to other features.
Using `netif` was the best choice but I realized it was already defined as a pseudomodule by the build system. And after looking a bit closer I realized it was not really useful.
So the 3 first commits of this PR are just removing the existing `netif` pseudomodule.

The rest of the commits are providing the `netif` to boards with an interface. The last 2 commits are just dropping the `BOARDS_PROVIDES_NETIF` variables from the related application and use `FEATURES_REQUIRED += netif` instead.

This is simpler to maintain and IMHO way cleaner than the actual design.





### Testing procedure

- A green Murdock
- Verify that boards providing the netif feature are listed in the results of (note that some new ones were added and some were removed):
```
$ make --quite -C examples/default info-boards-supported
$ make --quite -C tests/netstats_l2 info-boards-supported
```




### Issues/PRs references

None
